data/binding/printf.go

```diff
@@ -0,0 +1,214 @@
+package binding
+
+import (
+	"fmt"
+)
+
+type sprintfString struct {
+	String
+
+	format string
+	source []DataItem
+	err    error
+}
+
+// NewSprintf returns a String binding that format its content using the
+// format string and the provide additional parameter that must be other
+// data bindings.
+//
+// Since: 2.2
+func NewSprintf(format string, b ...DataItem) (String, error) {
+	ret := &sprintfString{String: NewString(), format: format}
+
+	for _, value := range b {
+		value.AddListener(ret)
+
+		ret.source = append(ret.source, value)
+	}
+
+	return ret, nil
+}
+
+func (s *sprintfString) DataChanged() {
+	var data []interface{}
+
+	s.err = nil
+	for _, value := range s.source {
+		switch x := value.(type) {
+		case Bool:
+			{
+				b, err := x.Get()
+				if err != nil {
+					s.err = err
+					return
+				}
+
+				data = append(data, b)
+			}
+		case Bytes:
+			{
+				b, err := x.Get()
+				if err != nil {
+					s.err = err
+					return
+				}
+
+				data = append(data, b)
+			}
+		case Float:
+			{
+				f, err := x.Get()
+				if err != nil {
+					s.err = err
+					return
+				}
+
+				data = append(data, f)
+			}
+		case Int:
+			{
+				i, err := x.Get()
+				if err != nil {
+					s.err = err
+					return
+				}
+
+				data = append(data, i)
+			}
+		case Rune:
+			{
+				r, err := x.Get()
+				if err != nil {
+					s.err = err
+					return
+				}
+
+				data = append(data, r)
+			}
+		case String:
+			{
+				str, err := x.Get()
+				if err != nil {
+					s.err = err
+					// Set error?
+					return
+				}
+
+				data = append(data, str)
+			}
+		case URI:
+			{
+				u, err := x.Get()
+				if err != nil {
+					s.err = err
+					return
+				}
+
+				data = append(data, u)
+			}
+		}
+	}
+
+	r := fmt.Sprintf(s.format, data...)
+	s.String.Set(r)
+}
+
+func (s *sprintfString) Get() (string, error) {
+	if s.err != nil {
+		return "", s.err
+	}
+	return s.String.Get()
+}
+
+func (s *sprintfString) Set(str string) error {
+	var data []interface{}
+
+	s.err = nil
+	for _, value := range s.source {
+		switch value.(type) {
+		case Bool:
+			data = append(data, new(bool))
+		case Bytes:
+			return fmt.Errorf("impossible to convert '%s' to []bytes type", str)
+		case Float:
+			data = append(data, new(float64))
+		case Int:
+			data = append(data, new(int))
+		case Rune:
+			data = append(data, new(rune))
+		case String:
+			data = append(data, new(string))
+		case URI:
+			return fmt.Errorf("impossible to convert '%s' to fyne.URI type", str)
```

Comment:
Maybe we could do `storage.ParseURI`?

Reply:
I think the limitation is I’m the current implementation not an actual limitation of URI string conversions. Just parse a string here and parse it to URI in the next code block?

Reply:
My idea for URI is logically to use them as a string, but as scanf requirement apply, I wasn't certain that all 'space' characters are properly escaped when generated and I wanted to keep this simple for introduction of the feature.

Reply:
Good point, but as an encoded URI it should not contain any spaces

Reply:
Ok, if we are sure that there is no space in fyne URI, then I should be able to process them as string. Will look at it.